src/components/EncounterBuilder/EncounterBuilder.jsx

```diff
@@ -1,10 +1,111 @@
+        return (<option value={monster.name}>{monster.name}</option>)
+      })
+      setMonsterList(monsterNames)
+    }
+  }, [allMonsters])
+
+
+  useEffect(() => {
+    console.log(filteredMonsters)
+  }, [filteredMonsters])
 
-const EncounterBuilder = () => {
   return (
-    <div className='EncounterBuilder'>
-      Encounter Builder Placeholder
+    <div className='encounter-builder'>
+      <section className='main-details'>
+        <section className='base-box'>
+          <section className='encounter-header'>
+            <h3>Encounter Name:</h3>
+            <input type="text" aria-label="encounter name input" required></input>
+            <div className='party-stats'>
+              <h3>Party Size:</h3>
+              <input type="text" aria-label="party size input" placeholder="" required></input>
+              <h3>Party Level:</h3>
+              <input type="text" aria-label="party level input" placeholder="" required></input>
+            </div>
+          </section>
+          <section className='encounter-desc'>
+            <h3>Short Summary:</h3>
+            <input type="text" aria-label="summary input" placeholder="" required></input>
+            <h3>Encounter Description:</h3>
+            <input type="text" aria-label="description input" placeholder="" required></input>
+            <h3>Treasure and Rewards:</h3>
+            <input type="text" aria-label="loot input" placeholder="" required></input>
+          </section>
+        </section>
+        <section className='encounter-foes base-box'>
+          <div className='monster-selection'>
+            <label htmlFor="size">Size:</label>
+            <select onChange={handleInputChangeSize} id='size'>
+              <option value=''></option>
+              <option value="Tiny">Tiny</option>
+              <option value="Small">Small</option>
+              <option value="Medium">Medium</option>
+              <option value="Large">Large</option>
+              <option value="Huge">Huge</option>
+              <option value="Gargantuan">Gargantuan</option>
+            </select>
+            <label htmlFor="name">Name:</label>
+            <select onChange={handleInputChangeName} id='name'>
+              <option value=""></option>
+              {monsterList}
+            </select>
+            <input type="number" aria-label="hit points search" placeholder="Hit Points" onChange={handleInputChangeHP}></input>
+            <input type="number"aria-label="armor class search" placeholder="Armor Class" onChange={handleInputChangeAC}></input>
+            <button onClick={() => {}}>Search</button>
+          </div>
+          <div className='monster-selection'>
+            <div>{filteredMonsters.map(monster => {
+              return (
+                <div className='monster-checker' key={monster.name}>
+                <p>{monster.name}</p>
+                <input type="checkbox" id={monster.name} name="name" value={monster.name}></input>
```

Comment:
I do not have 100% understanding. But the way the backend endpoint currently works, is we use monster index values for requests (the all downcased one with `-` instead of spaces).

Would this be using the monsterName attribute? Just wanted to make sure! If I am wrong, please ignore.

Reply:
It was `monster.name` while hooked up to mock data but I updated it to `monster.monsterName` with the live calls. Thanks for clarifying!